model-optimizer/extensions/front/tf/CTCLossReplacement.py

```diff
@@ -78,53 +59,23 @@ def pattern(self):
             ])
 
     def replace_sub_graph(self, graph: Graph, match: dict):
-        seq_len_tf = match['seq_len']
         transpose_tf = match['transpose']
         ctc_greedy_decoder_tf = match['ctc_greedy_decoder']
         cast_tf = match['cast']
         ctc_loss_tf = match['ctc_loss']
         sparse_to_dense_tf = match['sparse_to_dense']
+        ctc_data_permute = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0, 2])},
+                                                       {'name': ctc_greedy_decoder_tf.name + '/ctc_data_permute'})
+        ctc_data_permute.in_port(0).get_connection().set_source(transpose_tf.out_port(0))
```

Comment:
Change to `ctc_data_permute.in_port(0).connect(transpose_tf.out_port(0))` and move creation a Transpose node and connection to the if/else block in lines 93-96.

Reply:
Not moved because the Transpose used for transposing CTCGreedyDecoderSeqLen input data.